src/devices/Common/Iot/Device/Common/I2cRegisterAccess.cs

```diff
@@ -0,0 +1,120 @@
+﻿//// Licensed to the .NET Foundation under one or more agreements.
+//// The .NET Foundation licenses this file to you under the MIT license.
+//// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.I2c;
+using System.Runtime.InteropServices;
+
+namespace Iot.Device.Common
+{
+    /// <summary>
+    /// Implements low-level access to read and write to the provided I2C device.
+    /// Takes as generic parameter an <see cref="Enum"/> where each value represents a register
+    /// with its corresponding address, thereby forming a register map. Example:
+    /// <example>
+    /// <code>
+    /// enum RegisterMap : byte
+    /// {
+    ///     Register1 = 0x00,
+    ///     Register2 = 0x01,
+    ///     Register3 = 0x03
+    /// }
+    /// </code>
+    /// </example>
+    /// </summary>
+    public sealed class I2cRegisterAccess<TRegisterMap> : IDisposable
```

Comment:
this would be fine if all registers were a single byte. Some devices may have larger registers

Reply:
I wrote a comment about this in I2cRegisterAccess/I2cRegisterAccess<T>.
It seems to me that I2cDevice.Write() does not support register addresses larger than a byte since you have no way of telling the method how many bytes in the input ReadOnlySpan<byte> are register address and how many are data.

Reply:
I took a look at UnixI2cDevice, one of the implementations of I2cDevice, and there doesn't seem to be any references to register addresses or slicing of the data to write. OK, maybe I misunderstood and it is up to each device to know how many of the first bytes in the ReadOnlySpan constitute the register address - do you know if that is how register addressing works in this I2C implementation? If so, my I2cRegisterAccess classes are limited to my 8-bit register addressing use case.